File: src/jk_5/nailed/network/codec/CipherEncoder.java
package jk_5.nailed.network.codec;

import io.netty.buffer.ByteBuf;
import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.MessageToByteEncoder;
import io.netty.handler.codec.MessageToMessageEncoder;
import jk_5.nailed.network.CipherBase;

import javax.crypto.Cipher;
import java.util.List;

/**
 * No description given
 *
 * @author jk-5
 */
public class CipherEncoder extends MessageToByteEncoder<ByteBuf> {

    private final CipherBase cipher;

    public CipherEncoder(Cipher cipher){
        this.cipher = new CipherBase(cipher);
    }

    @Override
    protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) throws Exception{
        cipher.cipher(in, out);
    }
}

File: src/jk_5/nailed/network/codec/CipherEncoder.scala
package jk_5.nailed.network.codec

import io.netty.handler.codec.MessageToByteEncoder
import io.netty.buffer.ByteBuf
import jk_5.nailed.network.CipherBase
import javax.crypto.Cipher
import io.netty.channel.ChannelHandlerContext
import java.util

/**
 * No description given
 *
 * @author jk-5
 */
class CipherEncoder(private final val ci: CipherBase) extends MessageToByteEncoder[ByteBuf] {
  def this(cipher: Cipher) = this(new CipherBase(cipher))
  override def encode(ctx: ChannelHandlerContext, msg: ByteBuf, out: ByteBuf) = this.ci.cipher(msg, out)
}
